robottelo/hosts.py

```diff
@@ -135,13 +136,20 @@ def __init__(self, hostname, auth=None, **kwargs):
         elif isinstance(auth, str):
             # key file based authentication
             kwargs.update({'key_filename': auth})
+        self._satellite = kwargs.get('satellite')
         super().__init__(hostname=hostname, **kwargs)
 
+    @property
+    def satellite(self):
```

Comment:
I really dont like this approach where we are actually traversing back from the parent object to its child object! Infact, we should initiate host object and play with its properties in satellite class.

Reply:
I disagree. It is quite useful for a registered content host to have a reference to the satellite it is registered to. An example of that use is provided in this PR's change to the nailgun_host property.